### PR TITLE
Convert outstanding .js files to typescript

### PR DESCRIPTION
**Pull request by mvandenburgh**
Partly addresses #1008.

### CODE REVIEW

**Comment by mvandenburgh** [comment]
Netlify is reporting build failures for the deploy preview :thinking: .... I'll mark this as draft for now

File: src/rest.ts [hunk 1]
```diff
@@ -6,19 +6,25 @@ import {
...
+
 // Ensure contains trailing slash
 const dandiApiRoot = process.env.VUE_APP_DANDI_API_ROOT.endsWith('/')
   ? process.env.VUE_APP_DANDI_API_ROOT
   : `${process.env.VUE_APP_DANDI_API_ROOT}/`;
 
 const client = axios.create({ baseURL: dandiApiRoot });
 
-let oauthClient: OAuthClient|null;
+let oauthClient: OAuthClient|null = null;
```
Comment (dchiquito) [inline src/rest.ts:20]:
```suggestion
let oauthClient?: OAuthClient = null;
```
I think this is the magic syntax

**Review by dchiquito — APPROVED** [review]

File: src/rest.ts (hunk 1 above)
Comment (mvandenburgh) [reply src/rest.ts:20]: Unfortunately `| null` looks like the only way to do it. https://github.com/microsoft/TypeScript/issues/13321

**Comment by mvandenburgh** [comment]
Marking this as draft - I need to transfer this over to the dandi-api repo

File: src/rest.ts (hunk 1 above)
Comment (dchiquito) [reply src/rest.ts:20]: sad :'(

**Pull request closed without merging**